libs/application/core/src/validation/validators.ts

```diff
@@ -45,49 +44,37 @@ function partialSchemaValidation(
       ? originalSchema.pick({ [key]: true })
       : originalSchema
 
-    const answerIsArray = Array.isArray(answer)
-    const answerIsEmptyArray = Array.isArray(answer) && answer.length === 0
-    const answerIsObject = typeof answer === 'object' && !answerIsArray
-    const answerIsPrimitive = !answerIsArray && !answerIsObject
+    try {
+      trimmedSchema.parse({ [key]: answer })
+    } catch (e) {
+      error = populateError(error, e, newPath)
```

Comment:
I would think this would show an error message in the form when we have an array or object, is that not the case?

Reply:
It depends if the array or object is valid. 
Zod is able to check if an object/array/primitive is valid. 
So if we have an valid array/object then this will try to parse it and not throw an error and continue with life. 
But if the array/object is invalid then the function would iterate through nested values and see what items are invalid, then the error response would be something like this:
```
invalidArray: invalid input
invalidArray[${index}]: invalid input
invalidArray[${index}].invalidVarible: invalid input
```

So you get top down what items are invalid in the schema which means we can choice on which level we handle it in the template.
I tried to reflect the in the tests, but would be happy to add more tests if you have suggestions for any or add an comment if you think that would help 😊 
This function is not straight forward, so everything like this should help future devs with debugging/iterating on it.

Reply:
Awesome, looks good!